sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java

```diff
@@ -145,10 +146,19 @@ public FormRecognizerAsyncClient buildAsyncClient() {
         final FormRecognizerServiceVersion serviceVersion =
             version != null ? version : FormRecognizerServiceVersion.getLatest();
 
-        // Endpoint cannot be null, which is required in request authentication
-
         HttpPipeline pipeline = httpPipeline;
         // Create a default Pipeline if it is not given
+        pipeline = getDefaultHttpPipeline(buildConfiguration, pipeline);
+
+        final FormRecognizerClientImpl formRecognizerAPI = new FormRecognizerClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .build();
+
+        return new FormRecognizerAsyncClient(formRecognizerAPI, serviceVersion);
+    }
+
+    private HttpPipeline getDefaultHttpPipeline(Configuration buildConfiguration, HttpPipeline pipeline) {
```

Comment:
I feel the caller should check if the pipeline is null before calling into this.